ui/lib/src/nvui/chess.ts

```diff
@@ -448,32 +448,32 @@ export function possibleMovesHandler(yourColor: Color, cg: CgApi, variant: Varia
   };
 }
 
-const promotionRegex = /^([a-h]x?)?[a-h](1|8)=[kqnbr]$/;
-const uciPromotionRegex = /^([a-h][1-8])([a-h](1|8))[kqnbr]$/;
-const dropRegex = /^(([qrnb])@([a-h][1-8])|p?@([a-h][2-7]))$/;
+const promotionRegex = /^([a-hA-H]x?)?[a-hA-H](1|8)=[kqnbrKQNBR]$/;
+const uciPromotionRegex = /^([a-hA-H][1-8])([a-hA-H](1|8))[kqnbrKQNBR]$/;
+const dropRegex = /^(([qrnbQRNB])@([a-hA-H][1-8])|[pP]?@([a-hA-H][2-7]))$/;
 export type DropMove = { role: Role; key: Key };
 
 export function inputToMove(input: string, fen: string, chessground: CgApi): Uci | DropMove | undefined {
   const dests = chessground.state.movable.dests;
   if (!dests) return;
   const legalUcis = destsToUcis(dests),
     legalSans = sanWriter(fen, legalUcis),
-    cleaned = input.replace(/\+|#/g, '');
+    cleaned = input.replace(/\+|#/g, '').toLowerCase();
```

Comment:
Input must not be lowercased. You need to distinguish between `Bxc4` and `bxc4`. `sanToUci` will do that for you.

Reply:
Fixed ... hopefully